src/test/java/racingcar/domain/car/CarTest.java

```diff
@@ -0,0 +1,56 @@
+package racingcar.domain.car;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+import racingcar.domain.numberpicker.RandomNumberPicker;
+import racingcar.mock.FakeRandomNumberPicker;
+
+class CarTest {
+
+    @DisplayName("차를 처음 생성 시 전진 카운트는 0이다.")
+    @Test
+    void init() {
+        // when
+        Car car = Car.nameOf(CarName.from("haero"));
+
+        // then
+        assertThat(car.getForwardCount()).isEqualTo(0L);
+    }
+
+    @DisplayName("선택된 난수가 4~9 사이의 숫자라면 차는 한 칸 전진한다.")
+    @ParameterizedTest
+    @ValueSource(ints = {4, 5, 6, 7, 8, 9})
+    void moveForwardForNumberGreaterThanOrEqualTo4(int randomDigit) {
+        // given
+        RandomNumberPicker randomNumberPicker = new FakeRandomNumberPicker(randomDigit);
+
+        Car car = Car.nameOf(CarName.from("haero"));
+
+        // when
+        car.moveForward(randomNumberPicker);
+
+        // then
+        assertThat(car.getForwardCount()).isEqualTo(1L);
+    }
+
+    @DisplayName("선택된 난수가 0~3 사이의 숫자라면 차는 전진 하지 않는다.")
+    @ParameterizedTest
+    @ValueSource(ints = {0, 1, 2, 3})
+    void moveForwardForNumberLessThan4(int randomDigit) {
+        // given
+        RandomNumberPicker randomNumberPicker = new FakeRandomNumberPicker(randomDigit);
```

Comment:
```suggestion
        RandomNumberPicker randomNumberPicker = () -> randomDigit;
```
인터페이스에 메소드가 하나밖에 없으면 이렇게 반환값을 바로 지정해줄 수 있습니다!

Reply:
물론 인터페이스에 추상메서드가 하나만 있으면 문법적으로 `함수형 인터페이스`로 볼 수 있습니다. 또, 람다식으로 간결하게 구현하면 가독성도 높아지고요(약간 멋있는 것은 덤이고요 ㅋㅋ). 

그런데, 매번 이렇게 '구현'하는 것이 좋을 지는 조금 고민해볼만 합니다. 이 같은 구현이 TC마다 반복될 텐데, 
저는 `구현`이 `반복`되면 `중복`이라고 생각하거든요. 

또, 결국 중복을 줄일 수 있는 방법은 이미 해당 코드를 구현한 클래스를 만들어 재사용성을 높이는 것이 객체지향의 장점 중 하나라고 생각합니다. 이런 관점에서, 저는 일급 객체인 람다식을 쓸 때는 과연 정말로 유용할지 고민을 하고 적용하는 편입니당 :)